File: frontend/src/features/teacher/types/teacher.ts
export interface Teacher {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  speciality: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTeacherRequest {
  firstName: string;
  lastName: string;
  email: string;
  speciality: string;
}

export interface UpdateTeacherRequest {
  firstName: string;
  lastName: string;
  email: string;
  speciality: string;
}

export interface PaginatedTeachers {
  items: Teacher[];
  page: number;
  limit: number;
  total: number;
  totalPages: number;
}

File: backend/models/teacher.go
package models

import "time"

type Teacher struct {
	ID         int       `json:"id"`
	FirstName  string    `json:"firstName"`
	LastName   string    `json:"lastName"`
	Email      string    `json:"email"`
	Speciality string    `json:"speciality"`
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`
}
